src/protocol/mod.rs

```diff
@@ -2110,6 +2110,1191 @@ impl<'input> Request<'input> {
         }
         Ok(Request::Unknown(header, remaining))
     }
+    // Get the matching reply parser (if any) for this request
+    pub fn reply_parser<'a>(&self) -> Option<ReplyParsingFunction<'a>> {
```

Comment:
This prototype strikes me as odd. Do I understand it correctly as "the caller can choose *one* arbitrary lifetime and then she can parse buffers with that lifetime"?

I guess I should try using the function in my current `xtrace` trainwreck. That should make sure this is usable.

Edit: Ah, the next commit handles this with some `for<'a>`. That seems sensible. I'll leave this comment in anyway, but I guess it just became irrelevant.

Reply:
Yeah as you saw this got cleaned up later.